### PR TITLE
Has part

### PR DESCRIPTION
**Pull request by jeromeku**

### CODE REVIEW

File: datasets/has_part/README.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
+---
+YAML tags:
```
Comment (lhoestq) [inline datasets/has_part/README.md:2]:
```suggestion
```

File: datasets/has_part/README.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  },
+  "synset": {
+    "feature_type": "Sequence",
+    "feature": {
+      "feature_type": "Value",
+      "dtype": "string"
+    }
+  }
+}
+```
```
Comment (lhoestq) [inline datasets/has_part/README.md:111]: Here we expect a description of the fields in natural language.
Can you add a brief description for each field please ?

File: datasets/has_part/README.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+### Data Splits
+
```
Comment (lhoestq) [inline datasets/has_part/README.md:114]:
```suggestion

[More Information Needed]
```

File: datasets/has_part/has_part.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+class HasPart(datasets.GeneratorBasedBuilder):
+    @property
+    def manual_download_instructions(self):
+        return """\
+  You should download the dataset from https://allenai.org/data/haspartkb
+  Download hasPartKB.tsv to a dir of your choice,
+  which will be used as a manual_dir, e.g. `~/.manual_dir/has_part`
+  The data can then be loaded via:
+  `datasets.load_dataset("has_part", data_dir="~/.manual_dir/has_part")`.
```
Comment (lhoestq) [inline datasets/has_part/has_part.py:56]: Looks like the data can be downloaded without requiring manual download
Here is the url from allenai : 
- hasPartKB.tsv: https://drive.google.com/uc?export=download&id=1Ev4RqWcPsLI9rgOGAKh-_dFKqcEZ1u-G

You can use the dl_manager to download the resource as usual.

**Review by lhoestq — COMMENTED** [review]
Nice !

I also found a way to not require manual download, see my comment

File: src/datasets/commands/dummy_data.py [hunk 5]
```diff
@@ -83,6 +83,7 @@ def auto_generate_dummy_data_folder(
                 self.mock_download_manager.dummy_data_folder,
```
Comment (lhoestq) [inline src/datasets/commands/dummy_data.py:83]: can you remove the changes in this file please ?

Looks like `black` wanted to do some changes while it's actually fine (line length < 119)

File: datasets/has_part/has_part.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+
+        dl_fp = dl_manager.download_and_extract(_URL)
+        # input_fp = os.path.join(os.path.dirname(dl_fp), "hasPartKB.tsv")
+        # os.rename(dl_fp, input_fp)
```
Comment (lhoestq) [inline datasets/has_part/has_part.py:78]: can you remove the comments please ?

File: datasets/has_part/README.md (hunk 2 above)
Comment (lhoestq) [reply datasets/has_part/README.md:88]: Can you add a description of each field `arg1` `arg2` `score` `wikipedia_primary_page` `synset` ?

**Review by lhoestq — COMMENTED** [review]
Nice ! thanks for changing from manual download to direct download :)

File: datasets/has_part/has_part.py (hunk 6 above)
Comment (jeromeku) [reply datasets/has_part/has_part.py]: Done.  The README already has descriptions of each data field (lines 89 - 92) in the Data Fields section.

File: src/datasets/commands/dummy_data.py (hunk 5 above)
Comment (jeromeku) [reply src/datasets/commands/dummy_data.py]: Done

File: datasets/has_part/README.md [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+### Data Fields
+
+```json
+{'arg1': 'plant',
+ 'arg2': 'stem',
+ 'score': 0.9991798414303377,
+ 'synset': ['wn.plant.n.02', 'wn.stalk.n.02'],
+ 'wikipedia_primary_page': ['Plant']}
+
+```
```
Comment (lhoestq) [inline datasets/has_part/README.md:88]:
````suggestion

[More Information Needed]
```
{'arg1': 'plant',
 'arg2': 'stem',
 'score': 0.9991798414303377,
 'synset': ['wn.plant.n.02', 'wn.stalk.n.02'],
 'wikipedia_primary_page': ['Plant']}

```

### Data Fields

````

**Review by lhoestq — APPROVED** [review]
Awesome thank you !

**Comment by lhoestq** [comment]
merging since the CI is fixed on master

**Pull request merged**